src/vs/workbench/contrib/chat/browser/chat.contribution.ts

```diff
@@ -234,6 +238,50 @@ class ChatSlashStaticSlashCommandsContribution extends Disposable {
 	}
 }
 
+export class ChatUndockAction extends Action2 {
+
+	constructor() {
+		super({
+			id: 'chat.action.undock',
+			title: {
+				value: nls.localize('chatUndock', "Undock the panel chat (experimental)"),
+				mnemonicTitle: nls.localize({ key: 'miChatUndock', comment: ['&& denotes a mnemonic'] }, "&&Undock the panel chat (experimental)"),
+				original: 'Undock the panel chat (experimental)'
+			},
+			category: Categories.View,
+			precondition: ActiveEditorContext,
+			f1: true
+		});
+	}
+
+	override async run(accessor: ServicesAccessor): Promise<void> {
+
+		const chatWidgetService = accessor.get(IChatWidgetService);
+		const lastFocusedWidget = chatWidgetService.lastFocusedWidget;
+		if (!lastFocusedWidget) {
+			return;
+		}
+		const viewModel = lastFocusedWidget.viewModel;
+		if (!viewModel) {
+			return;
+		}
+		const sessionId = viewModel.sessionId;
+
+		const editorGroupService = accessor.get(IEditorGroupsService);
+		// create a chat editor and drop it inside of the group
+		const editorPanesRegistry = Registry.as<IEditorPaneRegistry>(EditorExtensions.EditorPane);
+		const instantiationService = accessor.get(IInstantiationService);
+		const chatEditorInput: ChatEditorInput = instantiationService.createInstance(ChatEditorInput, ChatEditorInput.getNewEditorUri(), { target: { sessionId } });
+		const editorPaneDescriptor = editorPanesRegistry.getEditorPane(chatEditorInput);
+		const auxiliaryEditorPart = await editorGroupService.createAuxiliaryEditorPart();
+		const editorPane = editorPaneDescriptor?.instantiate(instantiationService);
+		if (!editorPane || !editorPane.group || !editorPane.input) {
+			return;
+		}
+		editorPane.group.moveEditor(editorPane.input, auxiliaryEditorPart.activeGroup);
```

Comment:
Hm I was not expecting a `moveEditor` but just `openEditor` in the aux window?

Reply:
Hi @bpasero, it's for the moment a draft PR, the code currently does not work and I was planning to dig into why this is the case. But thanks for the tip, I'll have a look more closely at the `openEditor` method

Reply:
Sure thing, let me know if you need help.